xdsl/utils/lexer.py

```diff
@@ -42,13 +47,20 @@ def get_lines_containing(self,
                 return [source[start:next_start]], start, line_no
             while next_start < span.end:
                 next_start = source.find('\n', next_start + 1)
+                if next_start == -1:
+                    next_start = span.end
             return source[start:next_start].split('\n'), start, line_no
 
-    def at(self, i: int):
+    def at(self, i: int) -> str | None:
         if i >= self.len:
-            raise EOFError()
+            return None
```

Comment:
🙏  I've had issues with this thank you for fixing it

Reply:
Doesn't this break things currently?